app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+# Service to download ftp files from the serve
+class Post < ApplicationRecord
+  belongs_to :user, optional: true, counter_cache: :posts_counter
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  def recent_comments
+    comments.last(5)
+  end
+
+  def update_posts_counter
+    user.posts_counter = 0 if user.posts_counter.nil?
+    user.posts_counter += 1
+    user.save
+  end
```

Comment:
- [ ] The concept here is similar to the last two suggestions up there. Please work with the suggestion below, considering the Foreign Key of author_id that we talked about.
```suggestion
  def update_post_counter
    author.increment!(:posts_counter)
  end
```

  ### OPTIONAL SUGGESTION
  
  We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to the database, then make the `update_posts_counter` method private.

  ```ruby
    # app/models/post.rb
    after_save :update_posts_counter

    private

    def update_posts_counter
      author.increment!(:posts_counter)
    end
  ```